### PR TITLE
added tool tip(to code and open issues) and check language function

### PR DESCRIPTION
**Pull request by aj-ya**
1. I have replaced the default code icon(</>) to different language logos by implementing a language check function.
**Function_ Declaration**_(repoCard.js)_
`function displayLogo()`
    `{` 
    `if(languageLogoList.includes(repo.language))`
_//languageLogoList is imported from the .js file of the same name that contains the list of all logos available_ 
    `{`
      `var langLogo='https://raw.githubusercontent.com/aj-ya/github-covid-finder/pr/add_logos/src/images/icons/languages/'+repo.language.replace(/\s+/g,"_")+'.svg';`
      //loading the image form github .replace function to replace one or more whitespaces with a single underscore
      `return()`
**WIP:loading_ image from loacal file** 
     _//`var langLogo='../images/icons/languages/'+repo.language.replace(/\s+/g,"_")+'.svg';`_
      _//`return()`_
      _//`require()` fails when a dynamic path is passed._
      _//possible solution at  https://stackoverflow.com/questions/52109907/react-require-error-cannot-find-module_
      _//couldnt quite work it out_
    `}`
   ` else`
    `{`
      `return (<CodeIcon />)`
   ` }`
  `}`
**Function Call**
` <Text>`
          `<span class="tooltip" role="img" aria-label="language" style={{ verticalAlign: 'middle' }}>`
          `{displayLogo()}{checkNullLang()}`            
          `</span>`
        `</Text>`
//checkNullLang() is used to prevent blank tooltip to display when `repo.language` is null 
  `function checkNullLang()`
 ` {`
    `if(repo.language!=null)`
     ` return(<span class="tooltiptext">{repo.language}</span>)`
  `}`
**Result**
- All the languages in the 'githubLanguages' list have been added an icon. 
- [Language_logos](https://imgur.com/qxIrwG0)

2.  have added Tooltips(using css) for open issues and repo.language sections 
**CSS Code**_(layout.css)_
/*Tooltip container*/
`.tooltip {`
 `position: relative;`
  `display: inline-block;`
`}`
/*Tooltip text*/
`.tooltiptext {`
 ` visibility: hidden;`
  `width: 110px;`
  `bottom: 100%;`
  `left: 50%;`
 `margin-left: -60px;`
  `background-color: black;`
  `color: #fff;`
  `text-align: center;`
  `padding: 5px 0;`
  `border-radius: 6px;`
  `position: absolute;`
  `z-index: 1;`
`}`
/* Show the tooltip text when you mouse over the tooltip container */
`.tooltip:hover .tooltiptext {`
  `visibility: visible;`
`} `
**Result**
- [Open_Issues](https://imgur.com/MJ7zn49)
- [Language ToolTip](https://imgur.com/8wrc7Us.png)

### I'm an undergrad student and this is my first contribution to an open source project so suggest any corrections if needed and  please pardon any prior mistakes.

### CODE REVIEW

File: src/components/repoCard.js [hunk 1]
```diff
@@ -51,20 +52,46 @@ const RepoCard = ({ repo }) => {
...
+        </Text>
       </Grid>
     </Card>
   )
-}
+  function displayLogo()
+  { 
+    if(languageLogoList.includes(repo.language))
+    {
+      var langLogo='https://raw.githubusercontent.com/aj-ya/github-covid-finder/pr/add_logos/src/images/icons/languages/'+repo.language.replace(/\s+/g,"_")+'.svg?sanitize=true';//loading the image form github and not local files.replace function to replace one or more whitespaces with a single underscore
```
Comment (luisFilipePT) [inline src/components/repoCard.js]: why not a simple mapper?
(just pseudo code below)
```
const logoMapper = {
   javascript: 'path/to/js/logo',
   ...
```
or

```
const logoMapper = {
   javascript: <img width="16px" height="16px" src={jsLogo} alt="js" />,
   ...
```

🤔 

Thanks for the help, if you want to try these suggestions feel free, I'll try to take a better look on Monday, as is cannot approve but a great contribution

**Comment by luisFilipePT** [comment]
One more thing, if there is NO Icon lang we probably should display the icon and the string and not only the icon. 😄

File: src/components/repoCard.js (hunk 1 above)
Comment (aj-ya) [reply src/components/repoCard.js]: Thanks for the suggestion, will work on this today.

**Comment by aj-ya** [comment]
# Changes Made 
## 1. repoCard.js
``` 
function displayLogo()
  { 
    //check if logoMapper contains the repo.language as a key   
    if(logoMapper[repo.language]!==undefined)
    {
      return(<Text>
        <span class="tooltip" role="img" aria-label="language" style={{ verticalAlign: 'middle' }}>
        <img width="16px" height="16px" src={logoMapper[repo.language]} alt={repo.language} />
        {checkNullLang()}
        </span>
        </Text>
      )//checkNullLang() for tool tip visibility.
    }
    else
    {
   //just displaying code icon with repo.language with no tooltip 
      return (
      <Text>
        <span role="img" aria-label="code" style={{ verticalAlign: 'middle' }}>
          <CodeIcon />
        </span> 
        {repo.language}
      </Text>
        )
    }
  }
```
The else part just displays code icon and repo.language(if not NULL) as text... do you want that to have tooltip too?
## 2. languageLogoList.js
_Added path within require()_ 
```
export const logoMapper = {
    'Abap':require('../images/icons/languages/Abap.svg'),
    ...
}
```

**Comment by luisFilipePT** [comment]
@aj-ya please resolve conflicts so that I can test it 😄

**Comment by aj-ya** [comment]
the conflicting files were ` package.json ` and ` package-lock.json `. I've Resolved it. Have a look please.

**Comment by luisFilipePT** [comment]
@all-contributors please add @aj-ya  for code

**Pull request merged**